test/ILLink.RoslynAnalyzer.Tests/UnconditionalSuppressMessageCodeFixTests.cs

```diff
@@ -0,0 +1,512 @@
+	public sealed class RequiresAssemblyFilesAttribute : Attribute
+	{
+		public RequiresAssemblyFilesAttribute () { }
+		public string? Message { get; set; }
+		public string? Url { get; set; }
+	}
+}";
+		private const string usmDef = @"
+namespace System.Diagnostics.CodeAnalysis
+{
+	[AttributeUsage(AttributeTargets.All, Inherited = false, AllowMultiple = true)]
+    public sealed class UnconditionalSuppressMessageAttribute : Attribute
+    {
+        public UnconditionalSuppressMessageAttribute (string category, string checkId)
+		{
+			Category = category;
+			CheckId = checkId;
+		}
+		public string Category { get; }
+		public string CheckId { get; }
+		public string? Scope { get; set; }
+		public string? Target { get; set; }
+		public string? MessageId { get; set; }
+		public string? Justification { get; set; }
+	}
+}";
+
+
+		static Task VerifyUnconditionalSuppressMessageCodeFixWithRUC (
+			string source,
+			string fixedSource,
+			DiagnosticResult[] baselineExpected,
+			DiagnosticResult[] fixedExpected,
+			int? numberOfIterations = null)
+		{
+			var test = new VerifyCSUSMwithRUC.Test {
+				TestCode = source + rucDef + usmDef,
+				FixedCode = fixedSource + rucDef + usmDef,
+			};
+			test.ExpectedDiagnostics.AddRange (baselineExpected);
+			test.TestState.AnalyzerConfigFiles.Add (
+						("/.editorconfig", SourceText.From (@$"
+is_global = true
+build_property.{MSBuildPropertyOptionNames.EnableTrimAnalyzer} = true")));
+			if (numberOfIterations != null) {
+				test.NumberOfIncrementalIterations = numberOfIterations;
+				test.NumberOfFixAllIterations = numberOfIterations;
+			}
+			test.FixedState.ExpectedDiagnostics.AddRange (fixedExpected);
+			return test.RunAsync ();
+		}
+
+		static Task VerifyUnconditionalSuppressMessageCodeFixWithRAF (
+			string source,
+			string fixedSource,
+			DiagnosticResult[] baselineExpected,
+			DiagnosticResult[] fixedExpected,
+			int? numberOfIterations = null)
+		{
+			var test = new VerifyCSUSMwithRAF.Test {
+				TestCode = source + rafDef + usmDef,
+				FixedCode = fixedSource + rafDef + usmDef,
+			};
+			test.ExpectedDiagnostics.AddRange (baselineExpected);
+			test.TestState.AnalyzerConfigFiles.Add (
+						("/.editorconfig", SourceText.From (@$"
+is_global = true
+build_property.{MSBuildPropertyOptionNames.EnableSingleFileAnalyzer} = true")));
+			if (numberOfIterations != null) {
+				test.NumberOfIncrementalIterations = numberOfIterations;
+				test.NumberOfFixAllIterations = numberOfIterations;
+			}
+			test.FixedState.ExpectedDiagnostics.AddRange (fixedExpected);
+			return test.RunAsync ();
+		}
+
+
+		[Fact]
+		public async Task RequiresUnreferencedCodeDiagnosticFix ()
+		{
+			var test = @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [RequiresUnreferencedCode(""message"")]
+    public int M1() => 0;
+    int M2() => M1();
+}
+class D
+{
+    public int M3(C c) => c.M1();
+    public class E
+    {
+        public int M4(C c) => c.M1();
```

Comment:
A lot of tests with different visibility makes sense for RUC since it applies different things, but `UnconditionalSuppress` doesn't care about visibility, so I'm unclear on what this test is supposed to demonstrate.